pkg/epp/requestcontrol/plugins.go

```diff
@@ -57,3 +57,18 @@ type ResponseComplete interface {
 	plugins.Plugin
 	ResponseComplete(ctx context.Context, request *types.LLMRequest, response *Response, targetPod *backend.Pod)
 }
+
+// PrepareData is called by the director before scheduling requests.
+// PrepareData plugin is implemented by data producers which produce data from different sources.
+type PrepareData interface {
+	plugins.Plugin
+	PrepareData(ctx context.Context, request *types.LLMRequest, pods []types.Pod)
+}
+
+// AdmitRequest is called by the director after the PrepareData plugins and before scheduling.
+// AdmitRequest plugin is implemented by plugins for admission control. These plugins need to implement Admit method.
+type AdmitRequest interface {
```

Comment:
naming selection should be changed IMO.
more specifically, struct/interface names should be *nouns*, e.g., `AdmissionPlugin`, while function names should be *verbs* based, e.g., `AdmitRequest`.
additionally, the documentation should explain what happens when one is using multiple admission plugins (e.g., the request should pass all admission checks).